### PR TITLE
[backend/frontend]Test sms and emails in bulk

### PR DESCRIPTION
**Pull request by johanah29**

### CODE REVIEW

**Comment by johanah29** [comment]
#1107

File: openbas-api/src/main/java/io/openbas/rest/inject_test_status/InjectTestStatusApi.java [hunk 1]
```diff
@@ -24,14 +25,23 @@ public InjectTestStatus testInject(@PathVariable @NotBlank String injectId) {
...
-    return injectTestStatusService.findAllInjectTestsByExerciseId(exerciseId);
+  @PostMapping("/api/injects/bulk/test")
+  public List<InjectTestStatus> bulkTestInjects(@RequestBody List<String> injectIds) {
+    return injectTestStatusService.bulkTestInjects(injectIds);
   }
 
-  @GetMapping("/api/scenario/{scenarioId}/injects/test")
-  public List<InjectTestStatus> findAllScenarioInjectTests(@PathVariable @NotBlank String scenarioId) {
-    return injectTestStatusService.findAllInjectTestsByScenarioId(scenarioId);
+  @PostMapping("/api/exercise/{exerciseId}/injects/test")
```
Comment (RomuDeuxfois) [inline openbas-api/src/main/java/io/openbas/rest/inject_test_status/InjectTestStatusApi.java:33]: We have an ExerciseInjectApi and a ScenarioInjectApi to split the code

File: openbas-front/src/utils/Localization.js [hunk 2]
```diff
@@ -1186,6 +1186,11 @@ const i18n = {
       'Attribute mapping configuration': 'Configuration d\'attribut de mapper',
       'The imported file contains absolute dates (ex.: 9h30). A starting date must be provided for the Scenario to be build': 'Le fichier importé contient des dates absolues (ex.: 9h30). Une date de début doit être fournie pour que le scénario puisse être créé',
       Test: 'Test',
+      'Do you want to test this inject?':
```
Comment (RomuDeuxfois) [inline openbas-front/src/utils/Localization.js:1189]: Don't forget, we have the chineese language now :)

File: openbas-front/src/admin/components/simulations/simulation/injects/ExerciseInjects.tsx [hunk 3]
```diff
@@ -201,6 +203,17 @@ const ExerciseInjects: FunctionComponent<Props> = () => {
     injectContext.onBulkDeleteInjects(injectsToProcess.map((inject: Inject) => inject.inject_id));
   };
 
+  const massTestInjects = () => {
+    bulkTestInjects(injectsToProcess.map((inject: Inject) => inject.inject_id)).then((result: { data: InjectTestStatus }) => {
+      if (numberOfSelectedElements === 1) {
+        // @ts-expect-error Data is an array with one element
+        navigate(`/admin/exercises/${exercise.exercise_id}/tests/${result.data[0].status_id}`);
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/simulations/simulation/injects/ExerciseInjects.tsx:210]: To remove the ts-expect-error, I suppose the date type is InjectTestStatus[] ?

File: openbas-front/src/admin/components/scenarios/scenario/injects/ScenarioInjects.tsx [hunk 4]
```diff
@@ -170,6 +172,17 @@ const ScenarioInjects: FunctionComponent<Props> = () => {
     injectContext.onBulkDeleteInjects(injectsToProcess.map((inject: Inject) => inject.inject_id));
   };
 
+  const massTestInjects = () => {
+    bulkTestInjects(injectsToProcess.map((inject: Inject) => inject.inject_id)).then((result: { data: InjectStatus }) => {
+      if (numberOfSelectedElements === 1) {
+        // @ts-expect-error Data is an array with one element
```
Comment (RomuDeuxfois) [inline openbas-front/src/admin/components/scenarios/scenario/injects/ScenarioInjects.tsx:178]: To remove the ts-expect-error, I suppose the date type is InjectTestStatus[] ?

**Comment by RomuDeuxfois** [comment]
We don't have the same user workflow between bulk and popover action :
- bulk -> when I use bulk test, I am redirect to the test tab
- popover action -> when I use popover action, I have a link display at the top to go to the test tab

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**